pkg/operator/sync.go

```diff
@@ -39,8 +40,18 @@ func (optr *Operator) syncAll(config OperatorConfig) error {
 	}
 	glog.Info("Synched up cluster api controller")
 
-	return optr.syncStatus(v1.OperatorStatusCondition{
-		Type:    v1.OperatorStatusConditionTypeDone,
+	// TODO(alberto): create funcs to for atomic conds changes
```

Comment:
You can make the `syncStatus` to accept variable number of arguments, resp. a list with variable size.

Reply:
Yeh, this is a first step to have the right CRD object in place intentionally simple, hence the TODO. Once we get clarity on the CVO expectations for updgrades we can model some functions